tests/test_integrated_channels/test_cornerstone/test_exporters/test_learner_data.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for Cornerstone Learner Data exporters.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import datetime
+import unittest
+
+import ddt
+import mock
+from freezegun import freeze_time
+from pytest import mark
+
+from django.utils import timezone
+
+from integrated_channels.cornerstone.exporters.learner_data import CornerstoneLearnerExporter
+from test_utils import factories
+from test_utils.fake_catalog_api import setup_course_catalog_api_client_mock
+
+
+@mark.django_db
+@ddt.ddt
+class TestCornerstoneLearnerExporter(unittest.TestCase):
+    """
+    Tests of CornerstoneLearnerExporter class.
+    """
+
+    NOW = datetime.datetime(2017, 1, 2, 3, 4, 5, tzinfo=timezone.utc)
+    NOW_TIMESTAMP = 1483326245000
+
+    def setUp(self):
+        self.user = factories.UserFactory(username='CSOD1', id=1, email='csod1@email.com')
+        self.other_user = factories.UserFactory(username='CSOD2', id=2, email='csod2@email.com')
+        self.subdomain = 'fake-subdomain'
+        self.session_token = 'fake-session-token'
+        self.callback_url = 'http://fake-callback-url.com'
+        self.user_guid = "fake-guid"
+        self.course_id = 'course-v1:edX+DemoX+DemoCourse'
+        self.course_key = 'edX+DemoX'
+        self.enterprise_customer = factories.EnterpriseCustomerFactory()
+        self.enterprise_customer_user = factories.EnterpriseCustomerUserFactory(
+            user_id=self.user.id,
+            enterprise_customer=self.enterprise_customer,
+        )
+        self.data_sharing_consent = factories.DataSharingConsentFactory(
+            username=self.user.username,
+            course_id=self.course_id,
+            enterprise_customer=self.enterprise_customer,
+            granted=True,
+        )
+        self.config = factories.CornerstoneEnterpriseCustomerConfigurationFactory(
+            enterprise_customer=self.enterprise_customer,
+            active=True,
+        )
+        self.csod_transmission_audit = factories.CornerstoneLearnerDataTransmissionAuditFactory(
+            user_id=self.user.id,
+            session_token=self.session_token,
+            callback_url=self.callback_url,
+            subdomain=self.subdomain,
+            course_id=self.course_key,
+            user_guid=self.user_guid
+        )
+        course_catalog_api_client_mock = mock.patch('enterprise.api_client.discovery.CourseCatalogApiServiceClient')
+        self.course_catalog_client = course_catalog_api_client_mock.start()
+        setup_course_catalog_api_client_mock(self.course_catalog_client)
+        self.addCleanup(course_catalog_api_client_mock.stop)
+        super(TestCornerstoneLearnerExporter, self).setUp()
+
+    @ddt.data(NOW, None)
+    @freeze_time(NOW)
+    def test_get_learner_data_record(self, completed_date):
```

Comment:
we need add more units tests to assert `CornerstoneAPIClient` is hitting appropriate url for sending user's progress and assert it is sending all expected params.

Reply:
I guess you have added them already.